crates/noirc_frontend/src/hir/type_check/mod.rs

```diff
@@ -55,6 +58,22 @@ pub fn type_check_func(interner: &mut NodeInterner, func_id: FuncId) -> Vec<Type
         }
     }
 
+    let (expr_span, empty_function) =
+        if let HirExpression::Block(block) = interner.expression(function_body_id) {
+            let last_stmt = block.statements().last();
+            let span = last_stmt.and_then(|last| {
+                if let HirStatement::Expression(expr) = interner.statement(last) {
+                    interner.expr_span(&expr).into()
+                } else {
+                    None
+                }
+            });
+
+            (span, last_stmt.is_none())
+        } else {
+            (None, false)
```

Comment:
If the expression wasn't a block you can still grab its location via `interner.expr_location(function_body_id)`. We should default to this on line 68 as well.

Reply:
strange error occurs:

```
error: expected type Field, found type ()
  ┌─ /home/work/subject/oliver/noir/h/src/main.nr:3:4
  │
3 │ fn main() -> Field {
  │    ----      ----- expected Field because of return type
  │    │          
  │    () returned here
  ```

Reply:
I suppose the expression location for function_body_id is wrong. May need to check where the resolver sets this location

Reply:
Right now, the span of the function name is being saved. Should we change it to the body of the function?
```
FunctionKind::Normal => {
    let expr_id = self.intern_block(func.def.body);
    self.interner.push_expr_location(expr_id, func.def.span, self.file);
    HirFunction::unchecked_from_expr(expr_id)
}
```

```
FunctionDefinition {
    span: name.0.span(),
```

Reply:
this seems to break other diagnostics

Reply:
It seems odd that would affect other diagnostics. Lets file it as an issue for now and work on it some other time then